python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,258 @@
+
+    return s
+
+
+def is_tensor_type(typ: torch._C.Type):
+    return typ.isSubtypeOf(torch.TensorType.get()) or (
+        isinstance(typ, torch.OptionalType)
+        and typ.getElementType().isSubtypeOf(torch._C.TensorType.get())
+    )
+
+
+def build_script_function(
+    schema: torch._C.FunctionSchema,
+    args: List[torch._C.Argument],
+    kwargs: Dict[str, Any],
+) -> torch.jit.ScriptFunction:
+    """
+    Build a torch.jit.ScriptFunction that corresponds to the schema.
+    This is necessary so that we can pass a "hydrated" module to ModuleBuilder::importModule.
+    Hydrated meaning with inlined constants with correct values; a gotcha here is that if you don't inline the constants
+    then importModule assume default values (i think?) thereby (potentially) failing to support the op
+    (not sure how this makes sense but native_batch_norm for example doesn't work without train=False).
+
+    Alternatively, we can consider the constant values are similar to the shapes of the input tensors.
+    Just as by default we compile for every shape, we also compile for every set of non-tensor inputs.
+
+    Works by
+    1. creating an empty ts graph
+    2. creating and inserting a node corresponding to the op; nodes inserted are just names (hence following steps)
+    2. for each arg to the op:
+        a. if arg is a tensor: add input to the graph corresponding to arg
+           if arg is a constant (bool, int, list): inline a constant (at the top of the graph)
+        b. wire value from previous step to the op (ie SSA value corresponding to graph input is "passed" to op)
+        c. if arg is a tensor: fill in shape
+
+    Parameters
+    ----------
+    schema: torch._C.FunctionSchema (i.e., c10::FunctionSchema)
+        Function in TorchScript IR to turn into MLIR.
+    args: ...
+
+    """
+    graph = torch._C.Graph()
+    node = graph.insertNode(graph.create(schema.name, len(schema.returns)))
+    for i, arg in enumerate(schema.arguments):
+        kwarg = False
+        if arg.name in kwargs:
+            val = kwargs[arg.name]
+            kwarg = True
+        else:
+            val = args[i]
+
+        # if tensor then pass to op
+        if is_tensor_type(arg.type) and val is not None:
+            inp = graph.addInput()
+            if isinstance(arg.type, torch.OptionalType):
+                inp.setType(arg.type.getElementType())
+            else:
+                inp.setType(arg.type)
+
+            if kwarg:
+                inp.setDebugName(arg.name)
+        # if constant then inline
+        else:
+            inp = graph.insertConstant(val)
+            inp.node().moveBefore(node)
+
+        node.addInput(inp)
+
+    if node.hasMultipleOutputs():
+        for outp in node.outputs():
+            graph.registerOutput(outp)
+    else:
+        graph.registerOutput(node.output())
+
+    # this name will ultimately be the name of the func op in the mlir module and thus needs to be unique
```

Comment:
Please use proper capitalization and punctutation in the comments throughout this PR: https://llvm.org/docs/CodingStandards.html#commenting